pkg/cache/snapshot_test.go

```diff
@@ -1331,3 +1204,166 @@ func TestSnapshotAddRemoveWorkloadWithLendingLimit(t *testing.T) {
+			t.Errorf("unexpected flavors %s", diff)
+		}
+		if diff := cmp.Diff(getFlavors(clusterQueueSnapshot.RGByResource(corev1.ResourceMemory)), group1Expected); diff != "" {
+			t.Errorf("unexpected flavors %s", diff)
+		}
+		if diff := cmp.Diff(getFlavors(clusterQueueSnapshot.RGByResource("example.com/gpu")), group2Expected); diff != "" {
+			t.Errorf("unexpected flavors %s", diff)
+		}
+	})
+
+	t.Run("TestResourceGrouping", func(t *testing.T) {
+		snapshot := cache.Snapshot()
+		clusterQueueSnapshot := snapshot.ClusterQueues["test-clusterqueue"]
+
+		group1Expected := sets.New[corev1.ResourceName](corev1.ResourceCPU, corev1.ResourceMemory)
+		group2Expected := sets.New[corev1.ResourceName]("example.com/gpu")
+
+		if diff := cmp.Diff(clusterQueueSnapshot.RGByResource(corev1.ResourceCPU).CoveredResources, group1Expected); diff != "" {
+			t.Errorf("unexpected resources %s", diff)
+		}
+		if diff := cmp.Diff(clusterQueueSnapshot.RGByResource(corev1.ResourceMemory).CoveredResources, group1Expected); diff != "" {
+			t.Errorf("unexpected resources %s", diff)
+		}
+		if diff := cmp.Diff(clusterQueueSnapshot.RGByResource("example.com/gpu").CoveredResources, group2Expected); diff != "" {
+			t.Errorf("unexpected resources %s", diff)
+		}
+	})
+}
+
+func TestQuotaAndLimits(t *testing.T) {
+	cache := New(utiltesting.NewFakeClient())
+	cache.AddOrUpdateResourceFlavor(utiltesting.MakeResourceFlavor("on-demand").Obj())
+	cache.AddOrUpdateResourceFlavor(utiltesting.MakeResourceFlavor("spot").Obj())
+	cache.AddOrUpdateResourceFlavor(utiltesting.MakeResourceFlavor("gpu1").Obj())
+	cache.AddOrUpdateResourceFlavor(utiltesting.MakeResourceFlavor("gpu2").Obj())
+
+	t.Run("TestQuotasAndLimits", func(t *testing.T) {
+		clusterQueue := utiltesting.MakeClusterQueue("test-clusterqueue").
+			ResourceGroup(
+				*utiltesting.MakeFlavorQuotas("on-demand").
+					ResourceQuotaWrapper(corev1.ResourceCPU).NominalQuota("5").Append().
+					ResourceQuotaWrapper(corev1.ResourceMemory).NominalQuota("10").Append().
+					Obj(),
+				*utiltesting.MakeFlavorQuotas("spot").
+					ResourceQuotaWrapper(corev1.ResourceCPU).NominalQuota("0").BorrowingLimit("50").Append().
+					ResourceQuotaWrapper(corev1.ResourceMemory).NominalQuota("0").LendingLimit("100").Append().
+					Obj(),
+			).
+			ResourceGroup(
+				*utiltesting.MakeFlavorQuotas("gpu1").
+					ResourceQuotaWrapper("example.com/gpu").NominalQuota("0").BorrowingLimit("5").LendingLimit("10").Append().
+					Obj(),
+				*utiltesting.MakeFlavorQuotas("gpu2").
+					ResourceQuotaWrapper("example.com/gpu").NominalQuota("3").BorrowingLimit("1").LendingLimit("4").Append().
+					Obj(),
+			).
+			ClusterQueue
+
+		if err := cache.AddClusterQueue(context.Background(), &clusterQueue); err != nil {
+			t.Fatal("Failed to add CQ to cache", err)
+		}
+
+		snapshot := cache.Snapshot()
+		clusterQueueSnapshot := snapshot.ClusterQueues["test-clusterqueue"]
+
+		actualCapacitiesAndLimits := make(map[resources.FlavorResource]*ResourceQuota, 0)
```

Comment:
```suggestion
		gotCapacitiesAndLimits := make(map[resources.FlavorResource]*ResourceQuota, 0)
```
Excuse me a bit of annoyance, it is a non-blocking nit, and we are not consistent about it, but go generally tries to name variables short (see at the bottom [here](https://go.dev/wiki/CodeReviewComments)). 

So, want / got are more popular than expected / actual. This naming is also suggested in the style guides: https://google.github.io/styleguide/go/best-practices.html, or https://www.arp242.net/go-testing-style.html